chain/pool/txlist.go

```diff
@@ -82,25 +92,36 @@ func (nst *NonceSortedTxs) Push(tx *transaction.Transaction) error {
 
 	hash := tx.Hash()
 	nst.idx = append(nst.idx, hash)
+	if nst.txs == nil {
+		nst.txs = make(map[common.Uint256]*transaction.Transaction)
+	}
 	nst.txs[hash] = tx
 
 	return nil
 }
 
-func (nst *NonceSortedTxs) Pop() (*transaction.Transaction, error) {
+func (nst *NonceSortedTxs) PopN(n uint16) ([]*transaction.Transaction, error) {
 	nst.mu.Lock()
 	defer nst.mu.Unlock()
 
 	if nst.empty() {
-		return nil, ErrNonceSortedTxsEmpty
+		return []*transaction.Transaction{}, ErrNonceSortedTxsEmpty
```

Comment:
why not return `nil` here?

Reply:
In order to append the returnValue directly without any exception handler.

Reply:
Done in additional commit.